gocourse04/smart_zoo_test.go

```diff
@@ -0,0 +1,218 @@
+package main
+
+import (
+	"fmt"
+	"testing"
+)
+import "github.com/stretchr/testify/assert"
+
+func TestAddSector(t *testing.T) {
+	area := NewArea("area_1", "Happy birds", "Feathered")
+	sector := NewSector("sector_1", "Parrots", NewTechnicalRoom("tr_1"))
+	area.AddSector(sector)
+
+	s, exists := area.GetSector("sector_1")
+	assert.True(t, exists, "Sector should exist")
+	assert.Equal(t, sector, s, "Sectors should be equal")
+}
+
+func TestDeleteSector(t *testing.T) {
+	area := NewArea("area_1", "Happy birds", "Feathered")
+	sector := NewSector("sector_1", "Parrots", NewTechnicalRoom("tr_1"))
+	area.AddSector(sector)
+
+	_, exists := area.GetSector("sector_1")
+	assert.True(t, exists, "Sector shouldn't exist")
+
+	area.DeleteSector("sector_1")
+	_, exists = area.GetSector("sector_1")
+	assert.False(t, exists, "Sector shouldn't exist")
+}
+
+func TestAddAnimal(t *testing.T) {
+	sector := NewSector("sector_1", "Parrots", NewTechnicalRoom("tr_1"))
+	animal := NewAnimal("animal_1", "Frodo")
+	sector.AddAnimal(animal)
+
+	s, exists := sector.GetAnimal("animal_1")
+	assert.True(t, exists, "Animal should exist")
+	assert.Equal(t, animal, s, "Animals should be equal")
+}
+
+func TestDeleteAnimal(t *testing.T) {
+	sector := NewSector("sector_1", "Parrots", NewTechnicalRoom("tr_1"))
+	animal := NewAnimal("animal_1", "Frodo")
+	sector.AddAnimal(animal)
+
+	_, exists := sector.GetAnimal("animal_1")
+	assert.True(t, exists, "Animal should exist")
```

Comment:
Не потрібно перевіряти, чи існує тварина. Тому що ми ж тестуємо саме функцію `DeleteAnimal`.

```suggestion
	_, exists := sector.GetAnimal("animal_1")
	assert.True(t, exists, "Animal should exist")
```

Тести мають бути простими.